EventFiltering/PWGJE/fullJetFilter.cxx

```diff
@@ -0,0 +1,170 @@
+  {
+    // Bins here
+    Int_t nPtBins = 200;
+    Float_t kMinPt = 0.;
+    Float_t kMaxPt = 200.;
+    Int_t nPhiBins = 18 * 8;
+    Float_t kMinPhi = 0.;
+    Float_t kMaxPhi = 2. * TMath::Pi();
+    Int_t nEtaBins = 100;
+    Float_t kMinEta = -1.;
+    Float_t kMaxEta = 1.;
+
+    hProcessedEvents.setObject(new TH1I("hProcessedEvents", ";;Number of filtered events", kCategories, -0.5, kCategories - 0.5));
+
+    hEmcClusterPtEta.setObject(new TH2F("hEmcClusterPtEta", "Emc Clusters;#it{p}_{T};#eta", nPtBins, kMinPt, kMaxPt / 2, nEtaBins, kMinEta, kMaxEta));
+    hEmcClusterPtPhi.setObject(new TH2F("hEmcClusterPtPhi", "Emc Clusters;#it{p}_{T};#phi", nPtBins, kMinPt, kMaxPt / 2, nPhiBins, kMinPhi, kMaxPhi));
+    hSelectedClusterPtEta.setObject(new TH2F("hSelectedClusterPtEta", "Selected Clusters;#it{p}_{T};#eta", nPtBins, kMinPt, kMaxPt / 2, nEtaBins, kMinEta, kMaxEta));
+    hSelectedClusterPtPhi.setObject(new TH2F("hSelectedClusterPtPhi", "Selected Clusters;#it{p}_{T};#phi", nPtBins, kMinPt, kMaxPt / 2, nPhiBins, kMinPhi, kMaxPhi));
+    hEmcJetPtEta.setObject(new TH2F("hEmcJetPtEta", "Emc Jets;#it{p}_{T};#eta", nPtBins, kMinPt, kMaxPt, nEtaBins, kMinEta, kMaxEta));
+    hEmcJetPtPhi.setObject(new TH2F("hEmcJetPtPhi", "Emc Jets;#it{p}_{T};#phi", nPtBins, kMinPt, kMaxPt, nPhiBins, kMinPhi, kMaxPhi));
+    hSelectedJetPtEta.setObject(new TH2F("hEmcJetPtEta", "Emc Jets;#it{p}_{T};#eta", nPtBins, kMinPt, kMaxPt, nEtaBins, kMinEta, kMaxEta));
+    hSelectedJetPtPhi.setObject(new TH2F("hEmcJetPtPhi", "Emc Jets;#it{p}_{T};#phi", nPtBins, kMinPt, kMaxPt, nPhiBins, kMinPhi, kMaxPhi));
+  } // init()
+
+  // Declare filters
+  o2::aod::EMCALClusterDefinition clusDef = o2::aod::emcalcluster::getClusterDefinitionFromString(mClusterDefinition.value);
+  Filter clusterDefinitionSelection = o2::aod::emcalcluster::definition == static_cast<int>(clusDef);
+
+  Bool_t isJetInEmcal(aod::Jet const& jet)
+  {
+    double emcalEtaMin = -0.7, emcalEtaMax = 0.7, emcalPhiMin = 1.40, emcalPhiMax = 3.26; // Phi: 80 - 187 deg
+    double R = jet.r() * 1e-2;                                                            // Jet R is saved as round(100*R)
+    if ((jet.eta() >= emcalEtaMin + R) && (jet.eta() <= emcalEtaMax - R) && (jet.phi() >= emcalPhiMin + R) && (jet.phi() <= emcalPhiMax - R)) {
+      return true;
+    }
+    return false;
+  }
+
+  Bool_t isEvtSelected(double const& jetpt, double const& jeteta, double const& jetphi,
+                       double const& cluspt, double const& cluseta, double const& clusphi)
+  {
+    if (jetpt > f_jetPtMin && cluspt >= f_clusterPtMin) {
+      return true;
+    }
+    return false;
+  }
+
+  void process(soa::Join<aod::Collisions, aod::EvSels>::iterator const& collision,
+               aod::Jets const& jets,
+               selectedClusters const& clusters)
+  {
+    bool keepEvent[kCategories]{false};
+    double maxClusterPt = -1., maxSelectedJetPt = -1.;
+
+    if (!collision.alias()[kTVXinEMC]) {
```

Comment:
You must return `tags(keepEvent[0]);`. By the way, you should not return the collision id.